universal-application-tool-0.0.1/app/controllers/admin/AdminQuestionController.java

```diff
@@ -356,14 +360,19 @@ private void updateDefaultLocalizationForOptions(
     // translations. If we do not have existing translations for a given string, create
     // a new, empty set of translations.
     ImmutableList.Builder<QuestionOption> updatedOptionsBuilder = ImmutableList.builder();
-    for (int i = 0; i < updatedDefaultOptions.size(); i++) {
-      String updatedDefaultOption = updatedDefaultOptions.get(i);
-      if (existingTranslations.containsKey(updatedDefaultOption)) {
-        updatedOptionsBuilder.add(
-            existingTranslations.get(updatedDefaultOption).toBuilder().setId(i).build());
+    for (QuestionOption updatedOption : updatedOptions) {
+      if (existingTranslations.containsKey(updatedOption.optionText().getDefault())) {
+        QuestionOption existingOption =
+            existingTranslations.get(updatedOption.optionText().getDefault());
+        if (existingOption.id() == updatedOption.id()) {
+          updatedOptionsBuilder.add(
+              existingOption.toBuilder()
+                  .setId(updatedOption.id())
+                  .setDisplayOrder(updatedOption.displayOrder())
+                  .build());
+        }
```

Comment:
(blocking): if the default options match, but the IDs do not match, then this does nothing?

Is there a weird interaction here if something is "deleted", but then brought back? e.g. v1 has `[a, b, c]`, and then v2 deletes `a` and has `[b,c]` but then v3 adds a new option and names it `a`?

Reply:
Hmm, that _does_ behave oddly, doesn't it.  Fixed.